app/controllers/leaderboard.js

```diff
@@ -25,309 +24,230 @@ module.exports.syncAthletes = function(req, res, next) {
 };
 
 module.exports.createRanking = function(req, res, next) {
-  /* Men wod1 */
-  Athlete.find({sex: 'm'}).sort({wod1Score: -1, wod1TieBreak: 1}).exec(function(err, athletes) {
-    if (err) console.log(err);
-
-    var pos = 1, next = 1, prev = 0, prevTieBreak = 0, prevRank = 1;
-    async.eachSeries(athletes, function(athlete, callback) {
-      var rank = pos;
-
-      if (athlete.wod1Score == prev && athlete.wod1TieBreak == prevTieBreak) {
-        rank = prevRank;
-      } else {
-        next++;
-        prevRank = pos;
-        pos = next;
-      }
-
-      prev = athlete.wod1Score;
-      prevTieBreak = athlete.wod1TieBreak;
-
-      Athlete.update({id: athlete.id}, {$set: {wod1Rank: rank}}, function(err, result) {
-        if (err) console.log(err);
 
-        console.log('Updated rank of ' + athlete.name);
-        callback();
-      });
-    });
-  });
+  /* Men wod1 */
+  saveAthletesResults(
+    {sex: 'm', 'wod1IsScale': false, 'wod1TieBreak': {$ne:null}},
```

Comment:
To be more legible, you could pass only one JSON object to the function. Create new attributes to associate the current function parameters